### ISSUE TITLE
[BUG] Backprop tutorial has an incorrect gradient result compared to other built-in methods

### ISSUE THREAD

**Issue by isaacdevlugt** [PennyLaneAI/qml#1050]
Reported here: https://discuss.pennylane.ai/t/possible-discrepancy-in-the-quantum-gradients-with-backpropagation-tutorial/4236
Demo: https://pennylane.ai/qml/demos/tutorial_backprop/

The gradient calculated manually via the below doesn't match what `jax.grad` or `qml.gradients.param_shift` gives 🤔.

```
def parameter_shift_term(qnode, params, i):
    shifted = params.copy()
    shifted = shifted.at[i].add(np.pi/2)
    forward = qnode(shifted)  # forward evaluation

    shifted = shifted.at[i].add(-np.pi/2)
    backward = qnode(shifted) # backward evaluation

    return 0.5 * (forward - backward)
```
